docs/devel/offline-virtual-machine.md

```diff
@@ -0,0 +1,234 @@
+GET /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+DELETE /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+PUT /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+PATCH /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+```
+
+By **POST** you can store new object in the etcd. With **GET** you either
+get the list of all OfflineVirtualMachines or get the concrete one. **DELETE**
+removes the object from etcd and all its resources. If you want to update the
+existing OfflineVirtualMachine object use **PUT** and if you want to change
+an item inside the object use **PATCH**.
+More details on the API are in the [documentation](https://kubevirt-incubator.github.io/api-reference/content/operations.html).
+
+To data format used when communicating with the API is the JSON. The format is
+set up the ussual way by setting the Content-Type header to 'application/json'.
+The 'application/yaml' can also be used.
+
+## OfflineVirtualMachine object content
+
+Now its time to discuss the content of the OfflineVirtualMachine object.
+The object is defined in the same way as any other Kubernetes object. You can
+use YAML of JSON format to specify the content. The example structure is bellow:
+
+```yaml
+apiVersion: kubevirt.io/v1alpha1
+kind: OfflineVirtualMachine
+metadata:
+  name: myvm
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        my: label
+    spec:
+      domain:
+        resources:
+          requests:
+            memory: 8Mi
+        devices:
+          disks:
+          - name: disk0
+            volumeName: mypcv
+      volumes:
+        - name: mypvc
+          persistentVolumeClaim:
+            claimName: myclaim
+```
+
+The file specification follows the Kubernetes guide. The apiVersion is linked
+with the Kubevirt release cycle.
+
+In the metadata section, there is a *required* field, the **name**. Then
+following the spec section, there are two important parts. The **running**, which
+indicates the current state of the VirtualMachine attached to this VirtualMachine.
+Second is the **template**, which is the VirtualMachine template.
+
+Let us go over each of these fields.
+
+### Name
+
+The name if important for the OfflineVirtualMachine, it is used to find created
+VirtualMachine. The VirtualMachine name is directly derived from
+the OfflineVirtualMachine. This means, if OfflineVirtualMachine is names 'testvm',
+the VirtualMachine is named 'testvm' too.
+
+When the VirtualMachineStatefulSet is created from the OfflineVirtualMachine,
+its name is also derived. In the case of 'testvm', name of each virtual machine
+in the stateful set is prefixed with the 'testvm' and appended with the ordering:
+'testvm0', 'testvm1' and so on.
+
+### Template
+
+The template is directly the VirtualMachine object. It contains the metadata
+section and a full VirtualMachineSpec.
```

Comment:
Where `VirtualMachineSpec` is defined?

Reply:
Thank you for pointing it out. It is the same as the VirtualMachine. I will update the text so it is clear what is going on.